clang/lib/Driver/ToolChains/Clang.cpp

```diff
@@ -4676,12 +4676,21 @@ void Clang::ConstructJob(Compilation &C, const JobAction &JA,
     if (HasFPGA)
       CmdArgs.push_back("-fsycl-disable-range-rounding");
 
+    const auto &DeviceTriple = getToolChain().getTriple();
+    const auto &DeviceSubArch = DeviceTriple.getSubArch();
```

Comment:
Can you use `Triple` and `SubArchType` here instead of auto?

Reply:
Sure, still following this advice after moving the code into `InitPreprocessor.cpp`.